src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -152,6 +156,19 @@ public static void atomicWrite(Path targetPath, WriteCallback writeCallback) thr
         }
     }
 
+    public static void withIOWorker(Runnable task) {
+        saveDataTasks = saveDataTasks.thenRunAsync(task, Util.ioPool());
+    }
+
+    public static void clearWorkerWhenDone() {
```

Comment:
What is this method intended for? It's completely unused at the moment.

Reply:
Doh, forgot to gen the patches

Reply:
Then this should not be using iopool. It needs to be single threaded or 2 writes to the same file could be concurrent

Reply:
No, it just needs to not unnecessarily break the chain of futures. The futures themselves don't retain any reference to the runnables or any previous futures, so there isn't even any possibility of a memory leak and as such little value in cleaning up the future every time.